framework/CodeInterfaces/SERPENT/outputParser.py

```diff
@@ -0,0 +1,210 @@
+    parsed = parseLine(compLines[i])
+    # isotope as key, atomic density as value
+    compDict[parsed[0]] = parsed[1]
+
+  compDict = filterTrace(compDict, percentCutoff)
+  return compDict
+
+
+def searchKeff(resFile):
+  """
+  Searches and returns the mean keff value in the .res file.
+    @ In, resFile, string, path to .res file
+    @ Out, keffDict, dictionary, key = keff or sd
+                                 value = list of keff or sd
+  """
+  with open(resFile) as f:
+    lines = f.readlines()
+
+  keffList = []
+  sdList = []
+
+  for i in range(0, len(lines)):
+    if 'IMP_KEFF' in lines[i]:
+      keffList.append(keffLineParse(lines[i])[0])
+      sdList.append(keffLineParse(lines[i])[1])
+
+  keffDict = {}
+  keffDict['keff'] = keffList 
+  keffDict['sd'] = sdList
+  return keffDict
+
+
+def keffLineParse(keffLine):
+  """
+  Parses through the anaKeff line in .res file.
+    @ In, keffLine, string, string from .res file listing IMPKEFF
+    @ Out, keffTuple, tuple, (mean IMPKEFF, sd of IMPKEFF)
+  """
+  start = keffLine.find('=')
+  newKeffLine = keffLine[start:]
+  start = newKeffLine.find('[')
+  end = newKeffLine.find(']')
+  
+  # +3 and -1 is to get rid of leading and trailing whitespace
+  keffSd = newKeffLine[start + 3:end - 1]
+  (keff, sd) = keffSd.split(' ')
+  keffTuple = (keff, sd)
+  return keffTuple
+
+
+def csvRenderDict(csvFilename, dictionary, header):
+  """
+  Renders csv given the dictionary column 1 = key, column 2 = value.
+    @ In, csvFilename, string, path of csv file to be created
+    @ In, dictionary, dictionary, dictionary to be rendered into csv file
+    @ In, header, list, list of length 2 of header strings
+    @ Out, bool, bool, True if successful
+  """
+  with open(csvFilename, 'w') as csvFile:
+    writer = csv.writer(csvFile)
+    # write header
+    writer.writerow(header)
+    for key, value in dictionary.items():
+      writer.writerow([key, value])
+  return True
+
+def readFileIntoList(file):
+  """ 
+  Reads file into list, every line as element.
+    @ In, file, string, name of file
+    @ Out, listFromFile, list, contents in the file as list
+  """
+  read = open(file, 'r')
+  lines = read.readlines()
+  listFromFile = []
+  for line in lines:
+    listFromFile.append(line.strip())
+  read.close()
+  return listFromFile
+
+def findDeptime(inputFile):
+  """
+  Finds the deptime from the input file.
+    @ In, inputFile, string, input file path
+    @ Out, deptime, string, depletion time in days
+  """
+  hit = False
+  with open(inputFile, 'r') as file:
+    for line in file:
+      if line.split(' ')[0] == 'dep':
+        if line.split(' ')[1] != 'daystep':
+          print('Currently can only take daystep')
+          raise ValueError()
+        else:
+          hit = True
+          continue
+      if hit:
+        deptime = line.split(' ')[0]
+        break
+
+  return deptime
+
+
+
+def makeCsv(csvFilename, inBumatDict, outBumatDict,
+       keffDict, isoList, inputFile):
```

Comment:
this is plenty short to leave on one line (suggested raven standard is 120 characters)